app/cdap/components/DataPrep/DataPrepSidePanel/SamplingTab/index.js

```diff
@@ -0,0 +1,226 @@
+              />
+            </FormGroup>
+            {this.state.samplingOptions.includes('random') && (
+              <FormGroup check>
+                <Label check>
+                  <Input
+                    type="radio"
+                    name="sampling-options"
+                    value="random"
+                    defaultChecked={this.state.sampleType === 'random'}
+                  />
+                  {T.translate(`${PREFIX}.SampleTypes.random`)}
+                </Label>
+                <DescriptionTooltip
+                  description={T.translate(`${PREFIX}.SampleTypes.randomTooltip`)}
+                  placement="right"
+                />
+              </FormGroup>
+            )}
+            {this.state.samplingOptions.includes('stratified') && (
+              <FormGroup check>
+                <Label check>
+                  <Input
+                    type="radio"
+                    name="sampling-options"
+                    value="stratified"
+                    defaultChecked={this.state.sampleType === 'stratified'}
+                  />
+                  {T.translate(`${PREFIX}.SampleTypes.stratified`)}
+                </Label>
+                <DescriptionTooltip
+                  description={T.translate(`${PREFIX}.SampleTypes.stratifiedTooltip`)}
+                  placement="right"
+                />
+              </FormGroup>
+            )}
+          </FormGroup>
+          {this.state.sampleType === 'stratified' && (
+            <FormGroup onChange={this.setStrata}>
+              <Label for="strata-col-select">{T.translate(`${PREFIX}.strataPrompt`)}</Label>
+              <Input type="select" id="strata-col-select" defaultValue={this.state.strata}>
+                {this.state.headers.map((header) => {
+                  return (
+                    <option key={header} value={header}>
+                      {header}
+                    </option>
+                  );
+                })}
+              </Input>
+            </FormGroup>
+          )}
+          <FormGroup>
+            <Label for="limit">{T.translate(`${PREFIX}.numRowsPrompt`)}</Label>
+            <Input
+              id="limit"
+              value={this.state.limit}
+              onChange={this.setLimit}
+              onBlur={this.validateLimit}
+              type="number"
+              onKeyPress={(event) => {
+                // Only allow users to input whole integers
+                if (!/[0-9]/.test(event.key)) {
+                  event.preventDefault();
+                }
+              }}
+              min={1}
+              max={3000}
+            />
+          </FormGroup>
+          <FormGroup onChange={this.setTimeoutMs}>
+            <Label for="timeout-select">{T.translate(`${PREFIX}.TimeLimit.prompt`)}</Label>
+            <Input type="select" id="timeout-select" defaultValue={this.state.timeoutMs}>
+              <option value={-1}>{T.translate(`${PREFIX}.TimeLimit.Options.0`)}</option>
```

Comment:
It's better to not use a numeric flag value. You could make this the string `no-limit` and check for that in the handler.